src/mavsdk/core/include/mavsdk/component_type.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+namespace mavsdk {
+
+    /**
+     * @brief ComponentType of configurations, used for automatic ID setting
+     */
+    enum class ComponentType {
+        Autopilot, /**< @brief SDK is used as an autopilot. */
+        GroundStation, /**< @brief SDK is used as a ground station. */
+        CompanionComputer, /**< @brief SDK is used as a companion computer on board the MAV. */
+        Camera, /** < @brief SDK is used as a camera. */
+        Gimbal, /** < @brief SDK is used as a gimbal. */
+        Custom /**< @brief the SDK is used in a custom configuration, no automatic ID will be
+                  provided */
+    };
+
+} // namespace mavsdk
```

Comment:
newline

Reply:
Hmmm. Why? clang-format doesn't seem to be concerned. Oh, yes it does. It didn't fix it because I ran it before commiting, so the new file wasn't on the git list yet.

Reply:
Fixed and force pushed.